### PR TITLE
[Tracer] Setting parallelizeTestCollections to false to Triage CI Run

### PR DESCRIPTION
**Pull request by link04**
## Summary of changes
Setting `parallelizeTestCollections`  to `false` to test the outcome of running the tests like instead.

https://xunit.net/docs/running-tests-in-parallel#:~:text=xUnit.net%20v2-,MSBuild%20Runner,-The%20MSBuild%20runner

## Reason for change
There are multiple tests that are flaking in `master` by what seems to be an overloaded server, so trying to set this variable to see how the CI reacts on this PR.

### CODE REVIEW

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (5270)](https://github.com/DataDog/dd-trace-dotnet/tree/e53f56740bf9fa3460d90cd7b2a596361f75d4fc)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/21558a991a89bc7bef07514335faf62422fbc86e)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5270) - mean (74ms)  : 63, 85
     .   : milestone, 74,
    master - mean (74ms)  : 66, 82
     .   : milestone, 74,

    section CallTarget+Inlining+NGEN
    This PR (5270) - mean (985ms)  : 960, 1009
     .   : milestone, 985,
    master - mean (995ms)  : 967, 1024
     .   : milestone, 995,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5270) - mean (111ms)  : 107, 115
     .   : milestone, 111,
    master - mean (111ms)  : 107, 115
     .   : milestone, 111,

    section CallTarget+Inlining+NGEN
    This PR (5270) - mean (718ms)  : 695, 741
     .   : milestone, 718,
    master - mean (716ms)  : 695, 736
     .   : milestone, 716,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5270) - mean (94ms)  : 90, 99
     .   : milestone, 94,
    master - mean (94ms)  : 91, 97
     .   : milestone, 94,

    section CallTarget+Inlining+NGEN
    This PR (5270) - mean (666ms)  : 643, 690
     .   : milestone, 666,
    master - mean (675ms)  : 648, 701
     .   : milestone, 675,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5270) - mean (187ms)  : 182, 192
     .   : milestone, 187,
    master - mean (188ms)  : 183, 192
     .   : milestone, 188,

    section CallTarget+Inlining+NGEN
    This PR (5270) - mean (1,062ms)  : 1040, 1083
     .   : milestone, 1062,
    master - mean (1,050ms)  : 1026, 1074
     .   : milestone, 1050,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5270) - mean (271ms)  : 267, 276
     .   : milestone, 271,
    master - mean (269ms)  : 262, 275
     .   : milestone, 269,

    section CallTarget+Inlining+NGEN
    This PR (5270) - mean (866ms)  : 838, 894
     .   : milestone, 866,
    master - mean (860ms)  : 829, 891
     .   : milestone, 860,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5270) - mean (260ms)  : 256, 264
     .   : milestone, 260,
    master - mean (257ms)  : 252, 263
     .   : milestone, 257,

    section CallTarget+Inlining+NGEN
    This PR (5270) - mean (851ms)  : 827, 874
     .   : milestone, 851,
    master - mean (842ms)  : 817, 867
     .   : milestone, 842,

```

**Comment by andrewlock** [comment]
## Throughput/Crank Report:zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (5270)](https://github.com/DataDog/dd-trace-dotnet/tree/e1340883edc744ee70a6b0ddb053017fa71de6dc)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/21558a991a89bc7bef07514335faf62422fbc86e)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/89ae3f4ee7d85b824badda5873166707c9e560b4)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5270) (10.975M)   : 0, 10975298
    master (10.925M)   : 0, 10925303
    benchmarks/2.9.0 (10.620M)   : 0, 10620397

    section Automatic
    This PR (5270) (7.457M)   : 0, 7456987
    master (7.496M)   : 0, 7496085
    benchmarks/2.9.0 (7.702M)   : 0, 7701617

    section Trace stats
    This PR (5270) (7.790M)   : 0, 7790251
    master (7.872M)   : 0, 7871947

    section Manual
    This PR (5270) (9.482M)   : 0, 9482443
    master (9.635M)   : 0, 9635035

    section Manual + Automatic
    This PR (5270) (7.237M)   : 0, 7236890
    master (7.114M)   : 0, 7113688

    section Version Conflict
    This PR (5270) (6.383M)   : 0, 6383216
    master (6.292M)   : 0, 6292435

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5270) (9.635M)   : 0, 9634785
    benchmarks/2.9.0 (9.526M)   : 0, 9525834

    section Automatic
    This PR (5270) (6.781M)   : 0, 6780909

    section Trace stats
    This PR (5270) (6.753M)   : 0, 6753223

    section Manual
    This PR (5270) (8.338M)   : 0, 8337637

    section Manual + Automatic
    This PR (5270) (6.239M)   : 0, 6239429

    section Version Conflict
    This PR (5270) (5.686M)   : 0, 5686479

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5270) (10.230M)   : 0, 10229893
    master (9.856M)   : 0, 9856447
    benchmarks/2.9.0 (10.533M)   : 0, 10532864

    section Automatic
    This PR (5270) (7.347M)   : 0, 7347163
    master (7.076M)   : 0, 7076232
    benchmarks/2.9.0 (7.815M)   : 0, 7814762

    section Trace stats
    This PR (5270) (7.695M)   : 0, 7694594
    master (7.404M)   : 0, 7404111

    section Manual
    This PR (5270) (9.032M)   : 0, 9031617
    master (8.658M)   : 0, 8657799

    section Manual + Automatic
    This PR (5270) (7.171M)   : 0, 7170871
    master (6.832M)   : 0, 6831716

    section Version Conflict
    This PR (5270) (6.438M)   : 0, 6437631
    master (6.173M)   : 0, 6172929

```
```mermaid
gantt
    title Throughput Linux x64 (ASM) (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    master (7.559M)   : 0, 7559179
    benchmarks/2.9.0 (7.859M)   : 0, 7859139

    section No attack
    master (1.844M)   : 0, 1844300
    benchmarks/2.9.0 (3.196M)   : 0, 3196117

    section Attack
    master (1.451M)   : 0, 1451050
    benchmarks/2.9.0 (2.519M)   : 0, 2519149

    section Blocking
    master (3.102M)   : 0, 3101736

    section IAST default
    master (6.434M)   : 0, 6433840

    section IAST full
    master (5.626M)   : 0, 5625514

    section Base vuln
    master (0.917M)   : 0, 917115

    section IAST vuln
    master (0.881M)   : 0, 881291

```

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/MsTestV2Tests.cs [hunk 1]
```diff
@@ -56,6 +58,9 @@ public async Task SubmitTraces(string packageVersion)
                         // Check the span count
                         Assert.Equal(expectedSpanCount, spans.Count);
 
+                        var settings = VerifyHelper.GetCIVisibilitySpanVerifierSettings(packageVersion);
```
Comment (lucaspimentel) [inline tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/MsTestV2Tests.cs:61]: What does this have to do with running tests in parallel?

Comment (link04) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/MsTestV2Tests.cs:61]: So because of the errors I got for `Datadog.Trace.ClrProfiler.IntegrationTests.CI.MsTestV2Tests` and the PR I found from Tony working on the tests to change to snapshots(https://github.com/DataDog/dd-trace-dotnet/pull/5226) I tried merging the branch to mine to see if the tests didn't error out, but they still did... luckily Tony just replied on Slack that he is aware of the errors and looking into those.

Comment (tonyredondo) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/MsTestV2Tests.cs:61]: I did a rebase of my PR, can you do another take to check if that solves the issue?

Comment (tonyredondo) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/MsTestV2Tests.cs:61]: forget it, I need to rework on the snapshots... will do that tomorrow.

Comment (link04) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/MsTestV2Tests.cs:61]: Dale yo confio.

Comment (tonyredondo) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/MsTestV2Tests.cs]: My PR with the fixes should be passing now, you can test that branch to check if the issue is solved.
